src/DocRaptor.php

```diff
@@ -39,15 +40,17 @@ class ApiWrapper
     protected $baseurl;
 
 
-
     /**
+     * @param HttpTransferInterface $httpClient
      * @param string|null $api_key
      */
-    public function __construct($api_key = null)
+    public function __construct(HttpTransferInterface $httpClient, $api_key = null)
```

Comment:
Injecting the interface resp. any implementation of it is fine, I meant the api key param. And arrays are not objects in PHP. What I would do to make it user friendly for those who only want to instantiate one class is to make the dependency optional and then lazy instantiate it in the constructor.

    public function __construct(HttpTransferInterface $httpClient = null, $api_key = null)
    {
        if (is_null($httpClient) {
            $httpClient = new HttpClient();
        }
        $this->httpClient = $httpClient;
    }

If I don't have one really big gap in my knowledge, `__construct(SomeClass $someClass = new SomeClass())` is not possible in PHP.